modules/saml/lib/Auth/Source/SP.php

```diff
@@ -549,7 +551,7 @@ private function startSSO2(Configuration $idpMetadata, array $state)
         }
 
         if (isset($state['saml:NameID'])) {
-            if (!is_array($state['saml:NameID']) && !is_a($state['saml:NameID'], NameID::class)) {
+            if (!is_array($state['saml:NameID'])) {
```

Comment:
I believe it was a RedundantCondition... The state-array doesn't contain objects, or does it?

Reply:
I believe it does. E.g. metadata objects...

Reply:
Metadata as object? Isn't metadata always an array?
Anyway, I've reverted the change.. Let's see what the actual issue was

Reply:
The issue was actually a bug introduced in #1008.. The check for NameID objects was added, but the rest of the code was never prepared for objects instead of arrays.. Resolved!